Sources/Foundation/NSURLError.swift

```diff
@@ -79,6 +79,7 @@ public var NSURLErrorServerCertificateHasUnknownRoot: Int { return -1204 }
 public var NSURLErrorServerCertificateNotYetValid: Int { return -1205 }
 public var NSURLErrorClientCertificateRejected: Int { return -1206 }
 public var NSURLErrorClientCertificateRequired: Int { return -1207 }
+public var NSURLErrorServerCertificateWrongHost: Int { return -1208 }
```

Comment:
I do believe that it is ok to use `NSURLErrorServerCertificateUntrusted` instead of adding new kind of error. I know that [Apple documentation](https://developer.apple.com/documentation/foundation/1508628-url_loading_system_error_codes/nsurlerrorservercertificateuntrusted) for this constant says "A server certificate was signed by a root server that isn’t trusted". But these docs are notoriously incomplete.

There is another old-school [Technical Note TN2232](https://developer.apple.com/library/archive/technotes/tn2232/_index.html#//apple_ref/doc/uid/DTS40012884-CH1-SECTRUSTERRORS) which explains the details of server trust evaluation. It states that `NSURLErrorServerCertificateUntrusted` is general error telling us about trust evaluation issue. And one of reasons of evaluation fail is "DNS name mismatch". I think other kinds of error are provided for convenience, but if the issue is not so specific, you will get plain `NSURLErrorServerCertificateUntrusted`.

Reply:
Good point! Also less code to support